html/process/settings/user.php

```diff
@@ -125,6 +126,22 @@
         $user->settings['explicitFeatures'] = false;
     }
 
+    // Language settings
+    if (isset($_POST['userLanguage'])) {
```

Comment:
Might be worth checking for empty inputs here, i.e. `$_POST['userLanguage'] != ''` - it's a bit nitpicky, but saves some operations with `getLanguageNames()` when it's guaranteed to fail. Otherwise, looking good!